### PR TITLE
Adding equals() to Fragment, to facilitate comparisons in tests

### PR DESCRIPTION
**Pull request by mgoddard-pivotal**
Writing some unit tests for the PXF + S3 + Parquet code, and the equals() was necessary.  Also, added hashCode() to this class.

### CODE REVIEW

File: pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/Fragment.java [hunk 1]
```diff
@@ -140,4 +142,48 @@ public String getProfile() {
...
+		result = prime * result + Arrays.hashCode(metadata);
+		result = prime * result + ((profile == null) ? 0 : profile.hashCode());
+		result = prime * result + Arrays.hashCode(replicas);
+		result = prime * result + ((sourceName == null) ? 0 : sourceName.hashCode());
+		result = prime * result + Arrays.hashCode(userData);
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```
[the remark this answers is not in the document]
Comment (mgoddard-pivotal) [reply pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/Fragment.java]: Well, I just had my IDE, Eclipse, generate these two methods for me.  Line 161 does what you have listed there, your top line.  Line 165 performs the same check you have there as `if (o instancef Point) {`, then `obj` is cast as Fragment on line 167.  The remainder of the logic you have within the complex `... && ...` statement is equivalently expressed in the remaining checks, but within the `if ... else if ... else` clauses.  Is your comment meant to point out an issue with correctness, or more of an aesthetic issue?

[the remark this answers is not in the document]
Comment (mgoddard-pivotal) [reply pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/Fragment.java]: If it's the former, I should probably create a unit test to verify the behavior.  If it's the latter, could you point me to some precedents within this codebase?

[the remark this answers is not in the document]
Comment (shivzone) [reply pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/Fragment.java]: Not about correctness ... more about general java coding style. Not ideal to have too many condition checks with return statements

**Comment by lavjain** [comment]
https://ncrcoe.gitbooks.io/java-for-small-teams/content/specifics/100_know_how_to_implement_hashcode_and_equals.html

**Comment by mgoddard-pivotal** [comment]
Thanks for the feedback, folks!  And for that excellent reference, @lavjain.

**Comment by shivzone** [comment]
Nice reference @lavjain

**Comment by mgoddard-pivotal** [comment]
I will embed the check for Fragment equality into my test and leave this class, Fragment, alone.

**Pull request closed without merging**